### ISSUE TITLE
How can I specify what base rom to use for a patch, so that it loads it automatically?

### ISSUE THREAD

**Issue by chexo3** [SourMesen/Mesen#206]
Loading it manually each time is annoying, and my roundabout solution of using symlinks isn't pretty. I'd like to be able to name them different things, so I can know what rom hack I'm choosing from the menu. But there is no elegant solution.

**Comment by SourMesen** [comment]
There is no simple solution for this - IPS files themselves give absolutely no indication of what ROM they are supposed to be applied to.

In the next version, the File->Recent Files menu has been improved when it comes to IPS patches.
It will add a separate entry for a patched rom and list the IPS file's name, e.g:
"Super Mario Bros.nes [MyHack.ips]"

The "Recent Files" menu only keeps the last 10 games you've opened, however, so that might not be good enough based on how you use the emulator.

**Comment by GregoryMcGregerson** [comment]
Probably could create a database for patches and roms they work with and do MD5 hash matching. So when you open an IPS patch it loads the correct rom or gives an error stating you don't have it. But that would be a ton of testing to make sure it's accurate, although most of the info of which rom to use is on romhacking.net in the information section.

**Comment by chexo3** [comment]
Could there be some sort of naming convention for them? IE: [(U) Super Mario Bros.nes] SuperMarioUnlimited.ips? Or if you put them in a folder with the rom, it will assume the only rom in the folder is the base for all the patches. Perhaps if you get code a custom rom select menu in the feature they will show up as seperate games.

**Comment by SourMesen** [comment]
The best I could imagine would be having Mesen remember the location of IPS files you've used in the past (and on what roms) and have an option to display a dialog when loading the plain rom, asking you if you would want to apply one of the IPS files?

**Comment by SourMesen** [comment]
> Or if you put them in a folder with the rom, it will assume the only rom in the folder is the base for all the patches

I skipped over this idea when I first read this, I think - this was simple to implement, so I just did (f26601f80295101e87f1017c4445d5717a468d9d). So if you have a single rom in a folder with one or many ips/bps patches, loading any of the patches will automatically load that rom and apply the patch to the rom.  This only works with unzipped roms (e.g .nes, .fds or .unf files)

Hope that helps!
